src/Forms/Shared/Samples/Map/BrowseBuildingFloors/BrowseBuildingFloors.xaml.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright 2022 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using Esri.ArcGISRuntime;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Mapping.Floor;
+using Esri.ArcGISRuntime.Portal;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+namespace ArcGISRuntimeXamarin.Samples.BrowseBuildingFloors
+{
+    [ArcGISRuntime.Samples.Shared.Attributes.Sample(
+        name: "Browse building floors",
+        category: "Map",
+        description: "Display and browse through building floors from a floor-aware web map.",
+        instructions: "Use the combo box to browse different floor levels in the facility. Only the selected floor will be displayed.",
+        tags: new[] { "building", "facility", "floor", "floor-aware", "floors", "ground floor", "indoor", "level", "site", "story" })]
+    public partial class BrowseBuildingFloors : ContentPage
+    {
+        private const string _floorData = "https://ess.maps.arcgis.com/home/item.html?id=f133a698536f44c8884ad81f80b6cfc7";
+        private FloorManager _floorManager;
+
+        // Collection of floors.
+        private readonly Dictionary<string, FloorLevel> _floorOptions = new Dictionary<string, FloorLevel>();
+
+        public BrowseBuildingFloors()
+        {
+            InitializeComponent();
+            _ = Initialize();
+        }
+
+        private async Task Initialize()
+        {
+            try
+            {
+                // Gets the floor data from ArcGIS Online and creates a map with it.
+                Map map = new Map(new Uri(_floorData));
+
+                MyMapView.Map = map;
+
+                // Map needs to be loaded in order for floormanager to be used.
+                await MyMapView.Map.LoadAsync();
+                List<string> floorName = new List<string>();
+
+                // Checks to see if the layer is floor aware.
+                if (MyMapView.Map.FloorDefinition != null)
+                {
+                    await Application.Current.MainPage.DisplayAlert("Alert", "The layer is not floor aware.", "OK");
+                    return;
+                }
+
+                await MyMapView.Map.FloorManager.LoadAsync();
+
+                // Checks to see if the floormanager loaded and check to see if there is a floormanager.
+                if (MyMapView.Map.FloorManager.LoadStatus == LoadStatus.Loaded && MyMapView.Map.FloorManager != null)
```

Comment:
Since the `LoadAsync` call is awaited, I think you can count on any load errors to throw an exception, which means you don't need to check the load status here or in the `else if`. Also, with the order the two conditions are written, if the FloorManager is null here, you would get a `NullReferenceException` before the null check could happen.